filewatcher/filewatcher_test.go

```diff
@@ -283,3 +291,87 @@ func TestParserFailure(t *testing.T) {
 		t.Errorf("data.Get().(int64) expected %d, got %d", expected, value)
 	}
 }
+
+func limitedParser(t *testing.T, expectedSize int) filewatcher.Parser {
+	return func(f io.Reader) (interface{}, error) {
+		buf, err := ioutil.ReadAll(f)
+		if err != nil {
+			t.Error(err)
+			return nil, err
+		}
+		if len(buf) != expectedSize {
+			t.Errorf(
+				"Expected size of %d, got %d, data %q",
```

Comment:
:no_bell: nit: should we generally go for [lowercase error messages](https://github.com/golang/go/wiki/CodeReviewComments#error-strings)?

Reply:
The lower case requirement is only for the messages in constructed `error` values, not anything else.